src/core/ddsi/src/q_plist.c

```diff
@@ -382,14 +428,14 @@ static void fini_generic_embeddable (void * __restrict dst, size_t * __restrict
       case XQ:
         /* non-nested, so never a need to deallocate only some of the entries and no complications
            in locating the end of the sequence element description */
-        COMPLEX (XQ, ddsi_octetseq_t, {
+        COMPLEX (XQ, ddsi_octetseq_t, (void)0, {
```

Comment:
Two things: firstly, I suspect that it would be clearer if the macro expansion were done by hand here and all superfluous code were to be removed. Secondly, it is a bit unfortunate that it will end up processing trying to free each element, if it is a list of, say, integers. To some extent, that's unavoidable ... but there might be a nicer solution. Perhaps it would work out nicely if it were to return whether it had to any work and bail out early in this loop if there was nothing to be done.